choria/framework.go

```diff
@@ -269,25 +268,41 @@ func (fw *Framework) FederationMiddlewareServers() (servers srvcache.Servers, er
 
 // PuppetDBServers resolves the PuppetDB server based on configuration of _x-puppet-db._tcp
 func (fw *Framework) PuppetDBServers() (servers srvcache.Servers, err error) {
-	configured := ""
 	if fw.Config.Choria.PuppetDBHost != "" {
-		configured = fw.Config.Choria.PuppetDBHost
-		if fw.Config.Choria.PuppetDBPort != 0 {
-			configured += ":" + strconv.Itoa(fw.Config.Choria.PuppetDBPort)
-		}
-	}
+		configured := fmt.Sprintf("%s:%d", fw.Config.Choria.PuppetDBHost, fw.Config.Choria.PuppetDBPort)
 
-	if configured != "" {
 		servers, err = srvcache.StringHostsToServers([]string{configured}, "https")
 		if err != nil {
 			return servers, fmt.Errorf("could not parse configured PuppetDB host: %s", err)
 		}
-	}
+	} else {
```

Comment:
you can return servers above the else, then completely remove the else.  This removes the if within an else with more if's in it.  ends up reading a bit easier

Reply:
Hum, I though some code bellow all the condition was needed in all cases, but it is only useful in the context of SRV records querying.  I moved it next to the SRV lookup, and adjusted the conditions as requested.